File: benchmarks/php/class_dispatch.php
<?php
class Counter {
    public int $value;

    public function __construct(int $start) {
        $this->value = $start;
    }

    public function step(int $delta): int {
        $this->value = $this->value + $delta;
        return $this->value;
    }

    public function double_(): int {
        $this->value = $this->value * 2;
        return $this->value;
    }
}

$n = isset($argv[1]) ? (int) $argv[1] : 50000;

$c = new Counter(0);
$total = 0;

for ($i = 0; $i < $n; $i++) {
    if ($i % 100 === 0) {
        $c->value = $i;
        $total += $c->double_();
    } else {
        $total += $c->step(1);
    }
}

echo $total, PHP_EOL;

File: benchmarks/python/class_dispatch.py
import sys


class Counter:
    def __init__(self, start):
        self.value = start

    def step(self, delta):
        self.value = self.value + delta
        return self.value

    def double(self):
        self.value = self.value * 2
        return self.value


n = int(sys.argv[1]) if len(sys.argv) > 1 else 50_000

c = Counter(0)
total = 0

for i in range(n):
    if i % 100 == 0:
        c.value = i
        total += c.double()
    else:
        total += c.step(1)

print(total)
